### ISSUE TITLE
Task 7, Question 4: Do not use an edge detection and hough transform!!!

### ISSUE THREAD

**Issue by chulminy** [chulminy/CIVE497-CIVE700#102]
Here is a possible step to solve the problem. 

![CIVE497-700](https://user-images.githubusercontent.com/3227103/77311574-be50f800-6cd6-11ea-8117-52ca369381f8.png)

**Comment by k65yang** [comment]
Is the scale always constant? I am trying to do what you suggested but the measured length is always different depending on the identified zone from feature matching. The feature points are the same in both images (I checked).

![scale2](https://user-images.githubusercontent.com/58671575/77339012-8ca15680-6d01-11ea-816d-a6e5e04718b8.png)

But if I hard code the corners in, I get the right measurement. 
![scale1](https://user-images.githubusercontent.com/58671575/77339114-b65a7d80-6d01-11ea-87dc-0dcc2df60cea.png)


I checked my code so many times, I don't know where else I could be going wrong...

**Comment by chulminy** [comment]
How can the scale be changed? You are using a fixed size book cover template image (the image on the right in my drawing). You know the size of the booklet and the corresponding physical size. The ratio between them is the scale that I mentioned above. 

![image](https://user-images.githubusercontent.com/3227103/77345114-91b6d380-6d0a-11ea-816a-ff9a5bab7779.png)

You see the blue area. You estimated wrong homography matrix so your measurement is wrong!

Is it clear?

**Comment by k65yang** [comment]
Yes, I understand that the mistake has to lie with the homography matrix but I am just having a lot of trouble finding out how I am getting it wrong. 
If I hard code the corners in, then I get the correct answer. Meaning that my code to calculate the homography matrix is correct. But if I use the following points from feature matching, then I get the wrong answer. So it has to do with these points. 

```matlab
% both sets sorted clockwise
imgCoverFe = [24.0552   59.9572  192.5862   62.1607;
      165.9587  155.0447  214.1026  184.0450]; % points already scaled

imgTestFe = [499.9673  572.8519  792.8667  557.6646;
             561.2570  558.9533  703.9821  603.7714];
```

Nothing changes in the code except these points. The features that these points represent are also matching. Do you have any idea of where my source of error is?
